org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/util/nodeselection/NodeSelection.java

```diff
@@ -179,6 +211,10 @@ public static NodeSelection empty() {
      * @return the new instance
      */
     public static NodeSelection from(IStructuredSelection selection) {
+        if (selection instanceof NodeSelection) {
```

Comment:
If the argument `selection` is a `NodeSelection`, then the method brakes it's contract and doesn't create a new instance. I'd remove the if statement.